### PR TITLE
Update native configuration file best practices

### PR DESCRIPTION
**Pull request by galderz**
Placing custom configuration files in `META-INF/native-image/<group-id>/<artifact-id>` is the best way. It doesn't get in the way of what Quarkus generates, works out of the box without additional configuration and does not produce any native build warnings. Alternative methods have been removed to avoid confusion.

### CODE REVIEW

**Comment by geoand** [comment]
Is this related to https://github.com/quarkusio/quarkus/pull/38118?

**Comment by zakkak** [comment]
> Is this related to #38118?

Yes, IMO we need a combination of these two PRs.

**Comment by zakkak** [comment]
> This PR is the result of my own experiments.

@galderz sure, I think what @gsmet wanted to highlight here is that both PRs are touching the same issue. And IMO they should be combined to give the best result.

**Comment by geoand** [comment]
@zakkak exactly

**Review by Sanne — APPROVED** [review]

**Comment by galderz** [comment]
Hmm, looking at https://github.com/quarkusio/quarkus/pull/38118, the changes there are minimal and still suggests using an experimental flag that you get a warning with (`ResourceConfigurationFiles` et al). So I favour my changes instead of #38118.

**Review by zakkak — CHANGES_REQUESTED** [review]
As mentioned in https://github.com/quarkusio/quarkus/pull/38118#pullrequestreview-1840616344

@galderz it looks like @dliubars updated https://github.com/quarkusio/quarkus/pull/38118 and it now only includes the following changes (i.e. it no longer suggests using experimental options) which are currently missing from this PR:

1. replace references to resources-config.json with resource-config.json
2. replace references to reflection-config.json with reflect-config.json

I suggest you either:

1. merge https://github.com/quarkusio/quarkus/pull/38118 and rebase this PR
or
2. cherry-pick @dliubars commit (e911584bec477a80a3fb4ec86ead8ddbe1ce51f2) in your branch

**Comment by galderz** [comment]
I merged #38118 and I've now rebased this PR with the changes there.

File: docs/src/main/asciidoc/writing-native-applications-tips.adoc [hunk 1]
```diff
@@ -60,7 +60,9 @@ will include:
 ==== Using a configuration file
 
 If globs are not sufficiently precise for your use case and you need to rely on regular expressions, or if you prefer relying on the GraalVM infrastructure,
-you can also create a `resource-config.json` (the most common location is within `src/main/resources`) JSON file defining which resources should be included.
+you can also create a `resource-config.json` JSON file defining which resources should be included.
+Ideally this, and other native image configuration files, should be placed under the `src/main/resources/META-INF/native-image/<group-id>/<artifact-id>` folder.
+By placing them there, there is no additional configuration parameters that need to be passed into the native build.
```
Comment (zakkak) [inline docs/src/main/asciidoc/writing-native-applications-tips.adoc:65]:
```suggestion
This way they will be automatically parsed by the native build, without additional configuration.
```

**Review by zakkak — APPROVED** [review]
LGTM. Just added a suggestion that IMO makes the corresponding sentence a bit easier to read.

**Review by geoand — APPROVED** [review]

**Pull request merged**